stdlib/public/core/KeyPath.swift

```diff
@@ -1710,6 +1710,7 @@ internal struct RawKeyPathComponent {
     switch value {
     case .struct(let offset):
       var base2 = base
+      defer { _fixLifetime(base2) }
```

Comment:
I'm worried this issue might be lurking elsewhere too. Could we instead put the `_fixLifetime` in the implementation of `withUnsafeBytes` to ensure all uses have the pointee's lifetime extended to cover the closure?